rest_framework_json_api/renderers.py

```diff
@@ -378,6 +378,11 @@ def extract_included(
                         included_cache[new_item["type"]][
                             new_item["id"]
                         ] = utils.format_field_names(new_item)
+                        meta = cls.extract_meta(field, serializer_resource)
```

Comment:
Shouldn't this logic be moved into build_json_resource_obj? Even the formatting of the field names could be moved there. This is just at a quick glance maybe in practice there are more hurdles.

Reply:
I found some time to look into this and I was able to move the extract meta logic into the `build_json_resource_obj` and updated your PR.